### PR TITLE
Forecast and Flower Gift Implementation

### PR DESCRIPTION
**Pull request by amdevgwen**
- Added SpeciesFormChangeWeatherTrigger to allow for forms to change when weather does.
- Changed TrySetWeather and SummonPhases to trigger weather setting form Changes
- Preliminary support for ability suppression form triggers in SpeciesFormChangeAbilitySuppressionTrigger

Implemented:

- Forecast: Castform transforms with the weather.
- Flower Gift: Cherrim Transforms in the weather.

Not yet implemented:

- Both effects need further ability suppression triggers to reach full intended behavior.
- Currently they mirror generation 4 functionality
- SpeciesFormChangeAbilitySuppressionTrigger doesn't have a home yet

### CODE REVIEW

File: .gitignore
Comment (bennybroseph) [inline .gitignore:1]: What are the changes in the .gitignore for?

**Review by bennybroseph — CHANGES_REQUESTED** [review]
Needs TSDoc documentation on all new classes and functions and code comments where applicable.

File: .gitignore
Comment (danieland) [reply .gitignore:1]: The .vs/ folder is created for anyone using the Visual Studio IDE, it's just state files and the like: https://developercommunity.visualstudio.com/t/vs-folder/82249#T-N83453

**Comment by bennybroseph** [comment]
Closing this due to inactivity from the original author. Feel free to make a new request if interest is reignited

**Pull request closed without merging**